src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerFactoryExtensions.cs

```diff
@@ -43,12 +42,19 @@ public static class ApplicationInsightsLoggerFactoryExtensions
         /// </summary>
         /// <param name="factory"></param>
         /// <param name="filter"></param>
-        /// <param name="client">The instance of <see cref="TelemetryClient"/> to use for logging.</param>
+        /// <param name="serviceProvider">The instance of <see cref="IServiceProvider"/> to use for logging.</param>
         public static ILoggerFactory AddApplicationInsights(
             this ILoggerFactory factory,
-            TelemetryClient client,
+            IServiceProvider serviceProvider,
             Func<string, LogLevel, bool> filter)
         {
+            var client = serviceProvider.GetService<TelemetryClient>();
+            var debugLoggerControl = serviceProvider.GetService<DebugLoggerControl>();
+            if (debugLoggerControl != null)
+            {
+                debugLoggerControl.EnableDebugLogger = false;
```

Comment:
Idea here is to enable debug time logger only when there is no other logger added explicitly. So when adding debug logger we include  `debugLoggerControl.EnableDebugLogger` into it's filter statement https://github.com/Microsoft/ApplicationInsights-aspnetcore/pull/317/files#diff-ba05ccca641cf8f4dbb9accc9dc93780R37 and when you call `loggetFactory.AddApplicationInsights` `EnableDebugLogger ` get's set to `false` and debug logger disabled.